### ISSUE TITLE
Getting nested Highlights

### ISSUE THREAD

**Issue by kevdotbadger** [babenkoivan/scout-elasticsearch-driver#152]
The Highlight implementation doesn't allow names like 'contacts.name', due to it using the magic __get method.

Any idea how I can get nested highlighted data?

**Comment by Surt** [comment]
Yeah, it would be great to implement ```ArrayIterator``` or something. I will try to send a pull request soon.
